src/core/src/op/result.cpp

```diff
@@ -9,12 +9,72 @@
 #include <typeinfo>
 
 #include "itt.hpp"
+#include "openvino/core/descriptor_tensor.hpp"
 
 namespace ov {
+namespace descriptor {
+
+/**
+ * @brief Dedicated output descriptor for Result's output.
+ *
+ * The class holds Result's output specific names which are assigned tensor descriptor.
+ */
+class ResultOutput : public Output {
+public:
+    ResultOutput(Node* node, const size_t index, const std::shared_ptr<Tensor>& tensor)
+        : Output{node, index, tensor},
+          m_output_names{} {};
+
+    void set_names(const std::unordered_set<std::string>& names) override {
+        rm_tensor_output_names();
+        m_output_names = names;
+        get_tensor().add_names(m_output_names);
+    }
+
+    void add_names(const std::unordered_set<std::string>& names) override {
+        m_output_names.insert(names.begin(), names.end());
```

Comment:
Iteration over `std::unordered_set` might be slow.

Reply:
After some offline discussion and test it looks is there should not any performance penalty.